### PR TITLE
Middleware

### PR DESCRIPTION
**Pull request by masterjus**
* Create functionality for Middleware handle
* Documentation
* Tests

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -170,66 +215,96 @@ class ItemLogger
...
+
+class PaymentListener
+{
+    public array $middleware = [
+        FilterEntities::class,
+        `App\Listeners\RabbitEvents\Middleware\AnotherMiddleware@someAction`,
+    ];      
+
+    /** 
+     * @param string $event Event Name. Passing only fpr wildcard events
```
Comment (andrew-nuwber) [inline readme.md:282]:
```suggestion
     * @param string $event Event Name. Passing only for wildcard events
```

File: readme.md [hunk 2]
```diff
@@ -170,66 +215,96 @@ class ItemLogger
...
-$someString = 'Hello!';
 
-// Example 1. Old way to publish your data. Will be deprecated it next versions.
-// Remember: You MUST pass array of arguments
-fire('something.happened', [$model->toArray(), $someArray, $someString]);
+namespace App\Listeners\RabbitEvents\Middleware;
 
-// Example 2. Method `publish` from `Publishable` trait
-SomeEvent::publish($model, $someArray, $someString);
+class FilterEntities
```
Comment (andrew-nuwber) [inline readme.md:250]: What about an interface for middlewares? It has different behavior from Laravel's middlewares so it can be useful

Comment (masterjus) [reply readme.md:250]: the action method could be different (see example).

File: src/Dispatcher.php [hunk 3]
```diff
@@ -50,4 +50,90 @@ protected function getListenerClass($listener)
...
+
+            $throughMiddleware = $this->extractMiddleware($listener);
+
+            foreach ($throughMiddleware as $middleware) {
+                $result = $wildcard
+                    ? call_user_func($middleware, $event, ...array_values($payload))
+                    : call_user_func_array($middleware, $payload);
+
+                if (false === $result) {
+                    return null;
```
Comment (masterjus) [inline src/Dispatcher.php:72]: the question is: to return null or false to stop propagation for all other listeners?

Comment (masterjus) [reply src/Dispatcher.php:72]: @mrdzen @andrew-nuwber

Comment (andrew-nuwber) [reply src/Dispatcher.php:72]: I think it's better to keep default Laravel behavior

Comment (masterjus) [reply src/Dispatcher.php:72]: There's no analog in Laravel.

Comment (andrew-nuwber) [reply src/Dispatcher.php:72]: Got it, let's keep `null`

**Pull request merged**